app/src/main/java/com/mstudio/superheroarch/domain/GetEpisodeAndDetailsUseCase.kt

```diff
@@ -0,0 +1,52 @@
+package com.mstudio.superheroarch.domain
+
+import com.mstudio.superheroarch.presentation.EpisodeDetailsViewEntity
+import com.mstudio.superheroarch.presentation.SeasonAndEpisode
+import com.mstudio.superheroarch.presentation.mapToEpisodeDetailsViewEntity
+import com.mstudio.superheroarch.repository.RickAndMortyRepository
+import com.mstudio.superheroarch.repository.TmdbRepository
+
+class GetEpisodeAndDetailsUseCase(
+    private val repository: RickAndMortyRepository,
+    private val tmdbRepository: TmdbRepository
+) {
+    suspend operator fun invoke(episodeUrl: String): Result<EpisodeDetailsViewEntity> {
+        val episodeResult = repository.fetchEpisodeDetails(episodeUrl)
+        if (episodeResult.isSuccess) {
+            val episodeEntity = episodeResult.getOrThrow()
+
+            val seasonAndEpisode = extractSeasonAndEpisode(episodeEntity.episode)
+            val seasonNumber = seasonAndEpisode.seasonNumber ?: 0
+            val episodeNumber = seasonAndEpisode.episodeNumber ?: 0
+
+            val tmdbResult = tmdbRepository.getEpisodeTMDBInfo(seasonNumber, episodeNumber)
+
+            if (tmdbResult != null) {
+                val episodeDetailsViewEntity = mapToEpisodeDetailsViewEntity(
```

Comment:
change this map to be consistent with the rest of the project

Reply:
Solved in next commit